internal/pkg/xds/adsc/adsc.go

```diff
@@ -80,6 +80,15 @@ func (a *ADSC) Run() error {
 	return nil
 }
 
+func (a *ADSC) Restart() {
+	log.Infof("reconnecting to ADS server %s", a.cfg.DiscoveryAddr)
+	err := a.Run()
+	if err != nil {
+		log.Errorf("failed to Restart to ADS server %s: %v", a.cfg.DiscoveryAddr, err)
+		time.AfterFunc(reconnectDelay, a.Restart)
+	}
```

Comment:
Nit, you can scope the error like so:

```suggestion
	if err := a.Run(); err != nil {
		log.Errorf("failed to Restart to ADS server %s: %v", a.cfg.DiscoveryAddr, err)
		time.AfterFunc(reconnectDelay, a.Restart)
	}
```

Reply:
Maybe worth adding in the log that it will attempt reconnecting after `reconnectDelay`.

Reply:
Done.